spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigMapPropertySource.java

```diff
@@ -54,14 +54,15 @@ public Fabric8ConfigMapPropertySource(KubernetesClient client, String name) {
 	@Deprecated
 	public Fabric8ConfigMapPropertySource(KubernetesClient client, String name, String namespace,
 			Environment environment) {
-		super(getName(name, getApplicationNamespace(client, namespace)),
-				getData(client, name, getApplicationNamespace(client, namespace), environment, "", true));
+		super(getName(name, getApplicationNamespace(client, namespace, "Config Map", null)), getData(client, name,
+				getApplicationNamespace(client, namespace, "Config Map", null), environment, "", true));
```

Comment:
This is a little bit complicated to follow, but I will try to explain. 

 - We have a `public` constructor for `Fabric8ConfigMapPropertySource`, that takes a `namespace` as parameter, among other properties, which I do not care about here.
 - This constructor is called in `Fabric8ConfigMapPropertySourceLocator::getMapPropertySource`. This method internally calls `Fabric8ConfigUtils::getApplicationNamespace`; a method that will throw a `NamespaceResolutionFailedException` in case a namespace can not be found. 
 - So far nothing new.

Now, because `Fabric8ConfigMapPropertySource` is a `public` class that has a `public` constructor, someone can still call this one directly, of course. Callers, as such, can pass the namespace directly. What we used to do is call : 

```
	public static String getApplicationNamespace(KubernetesClient client, String namespace) {
		return !StringUtils.hasLength(namespace) ? client.getNamespace() : namespace;
	}
```

Get the namespace provided or from the client. So in one case we used to throw a `NamespaceResolutionFailedException`, but in the other one, we would end up querying all namespaces (see `Fabric8ConfigUtils::getConfigMapData`), in case the namespace was not present. 

---

Now we will call : `getApplicationNamespace(client, namespace, "Config Map", null)`, effectively doing the same logical thing:

- if `namespace` was provided, use that
- if it was not provided get it from the client
- notice that instead of `KubernetesNamespaceProvider`, we pass a `null`, in order to preserve the previous logic. 

The trick here  is that we would ned up throwing a `NamespaceResolutionFailedException` if a namespace is not present, instead of querying all namespaces (like we used to do). This, imho, also closes a bug, since we should not really query all namespaces (we have the same open defect for Secrets, which I will handle too).

Reply:
Can you point to the issue?

Reply:
there is no issue per-se, and can only happen if someone extends `Fabric8ConfigMapPropertySource` or the kubernetes-native property source, and do not provide a namespace. 

This PR makes such clients (as above) fail with the same `NamespaceResolutionFailedException` as we do in the usual flow. So : 

- plain users of our library + we can't resolve a namespace =>  `NamespaceResolutionFailedException`
- some clients that extends `Fabric8ConfigMapPropertySource/KubernetesClientConfigMapPropertySource` (for whatever reason) + we can't resolve a namespace => 

     a) in fabric8 we query all namespace to find such a configmap (not a great idea)
     b) in k8s-native, we would fail with a `ApiException`

This PR makes everything fail with the same `NamespaceResolutionFailedException`.

Reply:
OK this sentence made me think there was an issue open for this
>This, imho, also closes a bug, since we should not really query all namespaces (we have the same open defect for Secrets, which I will handle too).

To be honest I think that this change in behavior should only be made in main.

Reply:
snap! I thought I made it against `main`, as I agree with you. I will point it to `main`.

Reply:
now it's against `main`